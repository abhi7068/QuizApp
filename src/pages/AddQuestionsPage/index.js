import React, { useState } from 'react'
import { connect } from "react-redux";
import * as actions from '../../store/actions/actions'
import Input from '../../atoms/Input';
import ButtonSimple from '../../atoms/ButtonSimple';
import Container from '@material-ui/core/Container'
import Grid from '@material-ui/core/Grid'
import AddQuestionsForm from '../../organisms/AddQuestionForm';
import {useHistory} from "react-router-dom";
function AddQuestionsPage() {
    const history=useHistory();
   const goToQuiz=()=>{
       let path=`quiz`;
       history.push(path);
   }
    return (
        <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
        >
            <AddQuestionsForm />
            <Grid>
            <ButtonSimple onClick={goToQuiz}>
    Take quiz
</ButtonSimple>
</Grid>
        </Grid >
    )
}
const mapStateToProps = state => ({
    questions: state.quiz,
});



export default connect(
    mapStateToProps)(AddQuestionsPage)