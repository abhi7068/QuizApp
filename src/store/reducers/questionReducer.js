import * as actionTypes from "../actions/actionTypes";
const initialState = {
    quiz: [],
};
const questionReducer = (state = initialState, action) => {
    console.log(action);
    switch (action.type) {
        case actionTypes.ADD_QUESTION:
            console.log(action)
            return {
                ...state,
              quiz:[...state.quiz,action.question]
            };
        default:
            return state;
    }
};

export default questionReducer;