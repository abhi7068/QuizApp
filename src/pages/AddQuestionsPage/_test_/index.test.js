import React from 'react'
import AddQuestionsPage from '../../AddQuestionsPage/index'
import {render,cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import {createStore} from 'redux';
import {Provider} from 'react-redux';
import questionReducer from '../../../store/reducers/questionReducer'
import {BrowserRouter as Router,withRouter} from 'react-router-dom';
const store= createStore(questionReducer);
afterEach(cleanup);
it('matches snapshot',()=>{
const {asFragment} = render(
<Provider store={store}>
          <Router>
           <AddQuestionsPage />
        </Router>
      </Provider>
    );
expect(asFragment()).toMatchSnapshot();
});
 
it('it renders',()=>{
render(
<Provider store = {store}>
          <Router>
       <AddQuestionsPage/>
        </Router>
      </Provider>
  );
});


