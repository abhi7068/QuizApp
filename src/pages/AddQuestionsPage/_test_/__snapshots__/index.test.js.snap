// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
    >
      <form>
        <div
          class="MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated Mui-required Mui-required"
              data-shrink="false"
              for="content"
              id="content-label"
            >
              Add Question
              <span
                aria-hidden="true"
                class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
              >
                â€‰*
              </span>
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                data-testid="Input"
                id="content"
                name="question"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
        <br />
        <br />
        <div
          class="MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Add Option
            </span>
          </button>
        </div>
        <br />
        <br />
        <div
          class="MuiGrid-root"
        />
        <br />
        <br />
        <div
          class="MuiGrid-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
          </button>
          <br />
          <br />
        </div>
      </form>
    </div>
    <div
      class="MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Take quiz
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
